diff --git a/include/rclcpp/wait_set_template.hpp b/include/rclcpp/wait_set_template.hpp
--- a/include/rclcpp/wait_set_template.hpp
+++ b/include/rclcpp/wait_set_template.hpp
@@ -328,11 +328,12 @@ public:
     this->sync_remove_guard_condition(
       std::move(guard_condition),
       [this](std::shared_ptr<rclcpp::GuardCondition> && inner_guard_condition) {
-        inner_guard_condition->exchange_in_use_by_wait_set_state(false);
+        auto local_guard_condition = inner_guard_condition;
         // This method comes from the StoragePolicy, and it may not exist for
         // fixed sized storage policies.
         // It will throw if the guard condition is not in the wait set.
-        this->storage_remove_guard_condition(std::move(inner_guard_condition));
+        this->storage_remove_guard_condition(std::move(local_guard_condition));
+        inner_guard_condition->exchange_in_use_by_wait_set_state(false);
         if (this->wait_result_holding_) {this->wait_result_dirty_ = true;}
       });
   }
