--- a/battery_state_broadcaster/src/battery_state_broadcaster.cpp
+++ b/battery_state_broadcaster/src/battery_state_broadcaster.cpp
@@ -366,7 +366,8 @@
       if (interfaces.battery_percentage)
       {
         raw_battery_states_msg_.battery_states[i].percentage = static_cast<float>(
-          state_interfaces_[interface_cnt].get_optional<double>().value_or(kUninitializedValue));
+          state_interfaces_[interface_cnt].get_optional<double>().value_or(kUninitializedValue) /
+          100.0);
         sums_.percentage_sum += raw_battery_states_msg_.battery_states[i].percentage;
         interface_cnt++;
       }
@@ -379,7 +380,7 @@
         if (std::isfinite(min_volt) && std::isfinite(max_volt) && (max_volt > min_volt))
         {
           raw_battery_states_msg_.battery_states[i].percentage =
-            static_cast<float>((voltage - min_volt) * 100.0 / (max_volt - min_volt));
+            static_cast<float>((voltage - min_volt) / (max_volt - min_volt));
           sums_.percentage_sum += raw_battery_states_msg_.battery_states[i].percentage;
         }
         else
