--- a/nav2_planner/include/nav2_planner/planner_server.hpp
+++ b/nav2_planner/include/nav2_planner/planner_server.hpp
@@ -158,7 +158,7 @@
   template<typename T>
   void getPreemptedGoalIfRequested(
     typename nav2::SimpleActionServer<T>::SharedPtr & action_server,
-    typename std::shared_ptr<const typename T::Goal> goal);
+    typename std::shared_ptr<const typename T::Goal> & goal);
 
   /**
    * @brief Get the starting pose from costmap or message, if valid
--- a/nav2_planner/src/planner_server.cpp
+++ b/nav2_planner/src/planner_server.cpp
@@ -303,7 +303,7 @@
 template<typename T>
 void PlannerServer::getPreemptedGoalIfRequested(
   typename nav2::SimpleActionServer<T>::SharedPtr & action_server,
-  typename std::shared_ptr<const typename T::Goal> goal)
+  typename std::shared_ptr<const typename T::Goal> & goal)
 {
   if (action_server->is_preempt_requested()) {
     goal = action_server->accept_pending_goal();
