--- a/controller_interface/include/semantic_components/gps_sensor.hpp
+++ b/controller_interface/include/semantic_components/gps_sensor.hpp
@@ -68,7 +68,7 @@
     {
       return static_cast<int8_t>(data.value());
     }
-    return std::numeric_limits<int8_t>::max();
+    return sensor_msgs::msg::NavSatStatus::STATUS_UNKNOWN;
   }

   /**
@@ -83,7 +83,7 @@
     {
       return static_cast<uint16_t>(data.value());
     }
-    return std::numeric_limits<uint16_t>::max();
+    return sensor_msgs::msg::NavSatStatus::SERVICE_UNKNOWN;
   }

   /**
