--- a/force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp
+++ b/force_torque_sensor_broadcaster/src/force_torque_sensor_broadcaster.cpp
@@ -178,6 +178,7 @@
   param_listener_->try_get_params(params_);

   wrench_raw_.header.stamp = time;
+  wrench_raw_.header.frame_id = params_.frame_id;
   force_torque_sensor_->get_values_as_message(wrench_raw_.wrench);
   this->apply_sensor_offset(params_, wrench_raw_);
   this->apply_sensor_multiplier(params_, wrench_raw_);
@@ -194,6 +195,7 @@
     if (filtered && realtime_filtered_publisher_)
     {
       wrench_filtered_.header.stamp = time;
+      wrench_filtered_.header.frame_id = params_.frame_id;
       realtime_filtered_publisher_->try_publish(wrench_filtered_);
     }
   }
